diff --git a/node_modules/next/dist/esm/server/lib/implicit-tags.js b/node_modules/next/dist/esm/server/lib/implicit-tags.js
index 3294b8b..2359f7f 100644
--- a/node_modules/next/dist/esm/server/lib/implicit-tags.js
+++ b/node_modules/next/dist/esm/server/lib/implicit-tags.js
@@ -48,7 +48,7 @@ export async function getImplicitTags(page, pathname, fallbackRouteParams) {
     // Add the tags from the pathname. If the route has unknown params, we don't
     // want to add the pathname as a tag, as it will be invalid.
     if (pathname && (!fallbackRouteParams || fallbackRouteParams.size === 0)) {
-        const tag = `${NEXT_CACHE_IMPLICIT_TAG_ID}${pathname}`;
+        const tag = `${NEXT_CACHE_IMPLICIT_TAG_ID}${encodeURI(pathname)}`;
         tags.add(tag);
     }
     if (tags.has(`${NEXT_CACHE_IMPLICIT_TAG_ID}/`)) {
diff --git a/node_modules/next/dist/esm/server/web/spec-extension/revalidate.js b/node_modules/next/dist/esm/server/web/spec-extension/revalidate.js
index 865dc70..c171fc1 100644
--- a/node_modules/next/dist/esm/server/web/spec-extension/revalidate.js
+++ b/node_modules/next/dist/esm/server/web/spec-extension/revalidate.js
@@ -69,7 +69,7 @@ import { removeTrailingSlash } from '../../../shared/lib/router/utils/remove-tra
         console.warn(`Warning: revalidatePath received "${originalPath}" which exceeded max length of ${NEXT_CACHE_SOFT_TAG_MAX_LENGTH}. See more info here https://nextjs.org/docs/app/api-reference/functions/revalidatePath`);
         return;
     }
-    let normalizedPath = `${NEXT_CACHE_IMPLICIT_TAG_ID}${removeTrailingSlash(originalPath)}`;
+    let normalizedPath = `${NEXT_CACHE_IMPLICIT_TAG_ID}${encodeURI(removeTrailingSlash(originalPath))}`;
     if (type) {
         normalizedPath += `${normalizedPath.endsWith('/') ? '' : '/'}${type}`;
     } else if (isDynamicRoute(originalPath)) {
diff --git a/node_modules/next/dist/server/lib/implicit-tags.js b/node_modules/next/dist/server/lib/implicit-tags.js
index 30699c1..7b8c680 100644
--- a/node_modules/next/dist/server/lib/implicit-tags.js
+++ b/node_modules/next/dist/server/lib/implicit-tags.js
@@ -58,7 +58,7 @@ async function getImplicitTags(page, pathname, fallbackRouteParams) {
     // Add the tags from the pathname. If the route has unknown params, we don't
     // want to add the pathname as a tag, as it will be invalid.
     if (pathname && (!fallbackRouteParams || fallbackRouteParams.size === 0)) {
-        const tag = `${_constants.NEXT_CACHE_IMPLICIT_TAG_ID}${pathname}`;
+        const tag = `${_constants.NEXT_CACHE_IMPLICIT_TAG_ID}${encodeURI(pathname)}`;
         tags.add(tag);
     }
     if (tags.has(`${_constants.NEXT_CACHE_IMPLICIT_TAG_ID}/`)) {
diff --git a/node_modules/next/dist/server/web/spec-extension/revalidate.js b/node_modules/next/dist/server/web/spec-extension/revalidate.js
index 86f0a12..d4bd4a6 100644
--- a/node_modules/next/dist/server/web/spec-extension/revalidate.js
+++ b/node_modules/next/dist/server/web/spec-extension/revalidate.js
@@ -82,7 +82,7 @@ function revalidatePath(originalPath, type) {
         console.warn(`Warning: revalidatePath received "${originalPath}" which exceeded max length of ${_constants.NEXT_CACHE_SOFT_TAG_MAX_LENGTH}. See more info here https://nextjs.org/docs/app/api-reference/functions/revalidatePath`);
         return;
     }
-    let normalizedPath = `${_constants.NEXT_CACHE_IMPLICIT_TAG_ID}${(0, _removetrailingslash.removeTrailingSlash)(originalPath)}`;
+    let normalizedPath = `${_constants.NEXT_CACHE_IMPLICIT_TAG_ID}${encodeURI((0, _removetrailingslash.removeTrailingSlash)(originalPath))}`;
     if (type) {
         normalizedPath += `${normalizedPath.endsWith('/') ? '' : '/'}${type}`;
     } else if ((0, _utils.isDynamicRoute)(originalPath)) {
